### PR TITLE
command and query bus handlers as functions

### PR DESCRIPTION
**Pull request by sergigp**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 organization := "com.sergigp.quasar"
 name := "quasar"
 
-version := "0.1.1"
+version := "0.2"
```
Comment (rgomezcasas) [inline build.sbt:4]: This is a breaking change, we should increase a version

File: src/test/scala/com/sergigp/quasar/test/AsyncQueryBusTest.scala [hunk 2]
```diff
@@ -33,30 +30,30 @@ class AsyncQueryBusTest extends TestCase {
...
+      queryBus.subscribe[FindDummyUserQuery](QueryHandlers.dummyHandler(userFinder))
 
       val userId = UuidStringStub.random
 
       shouldNotFindUser(userId)
 
-      queryBus.ask(FindDummyUserQuery(userId)).futureValue.left.value should be (DummyUserNotFoundError(userId))
+      val result = queryBus.ask(FindDummyUserQuery(userId))
+
+      ScalaFutures.whenReady(result.failed) { e =>
```
Comment (rgomezcasas) [inline src/test/scala/com/sergigp/quasar/test/AsyncQueryBusTest.scala]: `error`?

File: src/main/scala/com/sergigp/quasar/command/AsyncCommandBus.scala [hunk 3]
```diff
@@ -1,40 +1,33 @@
...
 
-class AsyncCommandBus(logger: Logger) extends CommandBus[Future] {
+class AsyncCommandBus[C <: Command](logger: Logger) extends CommandBus[Future, C] {
 
-  private var handlers: Map[Class[_], CommandHandler[Future, _ <: Command]] = Map.empty
+  private var handlers = Map.empty[Class[_], C => Future[Unit]]
 
-  override def publish[C <: Command](
-    command: C
-  ): Future[Either[C#CommandError, Unit]] =
```
Comment (rgomezcasas) [inline src/main/scala/com/sergigp/quasar/command/AsyncCommandBus.scala:17]: Why did you removed the `either`?

File: build.sbt (hunk 1 above)
Comment (sergigp) [reply build.sbt:4]: if version < 1.0 == no estable

Comment (rgomezcasas) [reply build.sbt:4]: Our love is stable <3

File: src/test/scala/com/sergigp/quasar/test/dummy/modules/dummyuser/application/find/QueryHandlers.scala [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import com.sergigp.quasar.test.dummy.modules.dummyuser.application.DummyUserResponse
+import com.sergigp.quasar.test.dummy.modules.dummyuser.application.find.FindDummyUserError.FindDummyUserError
+import com.sergigp.quasar.test.dummy.modules.dummyuser.domain.find.UserFinder
+
+object QueryHandlers {
+  def dummyHandler(userFinder: UserFinder): FindDummyUserQuery => Future[Either[FindDummyUserError, DummyUserResponse]] =
+    (q: FindDummyUserQuery) => userFinder.find(q.id).map {
```
Comment (rgomezcasas) [inline src/test/scala/com/sergigp/quasar/test/dummy/modules/dummyuser/application/find/QueryHandlers.scala:12]: query?

**Review by rgomezcasas — APPROVED** [review]

**Pull request merged**